flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobServer.java

```diff
@@ -354,10 +354,12 @@ public void close() throws IOException {
         if (shutdownRequested.compareAndSet(false, true)) {
             Exception exception = null;
 
-            try {
-                this.serverSocket.close();
-            } catch (IOException ioe) {
-                exception = ioe;
+            if (serverSocket != null) { // can be null if shut down before constructor completion
```

Comment:
we would need another `null` check in further down in the `closeAsync` method (where the log message is created).

Reply:
Right, added check